lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -953,7 +963,17 @@ ParseResult FIRParser::parseType(FIRRTLType &result, const Twine &message) {
     if (forceable && innerType.containsConst())
       return emitError(loc, "rwprobe cannot contain const");
 
-    result = RefType::get(innerType, forceable);
+    SymbolRefAttr layer;
+    if (!layers.empty()) {
+      auto nestedLayers =
+          llvm::map_range(ArrayRef(layers).drop_front(), [&](StringRef a) {
+            return FlatSymbolRefAttr::get(getContext(), a);
+          });
+      layer = SymbolRefAttr::get(getContext(), layers.front(),
+                                 llvm::to_vector(nestedLayers));
+    }
+
+    result = RefType::get(innerType, forceable, layer);
```

Comment:
I can include error tests. I omitted them here as they aren't particularly interesting. E.g., there's really only one error case to check:

1. `Probe<UInt<1>, A`

This increases code coverage and is probably good to test.

WDYT?

Reply:
And it looks like this test should have been there before and was missing. 🙃 I'll add it.

Reply:
Test added in dea1a8e379dca787c4b9fe19a1baf9b0ed45c6be.

Reply:
Answering your larger question... I generally structure new features into several distinct PRs:

1. Add basic dialect support.
2. Add parser/emitter support.
3. Add verifier support (possibly as part of 1, but you usually miss stuff in 1 anyway and wind up adding stuff here).
4. Add pass support, likely broken into multiple PRs adding support in FIRRTL, then in HW/SV. This may also be broken up to handle on a per-pass basis if it's complex.
5. End-to-end work if not handled in (4).
6. Chisel API added.

(1) is a prerequisite of (2). Having (2) means you can start writing FIRRTL examples and make sure things work with the spec / iterate on the spec. (3--5) are then necessary to close it out.

As long as you don't add a Chisel API and a new feature is not load bearing (replacing something existing), then there's no need to land this all at once.

I'm going off this playbook here to try and make things easy to review. There is also my strong desire to not duplicate parsing checks in verifiers. If these are duplicated then you wind up having to keep both up to date. I frequently forget about it and things are out of date.

Do you agree with this approach?

Reply:
> What version should gate layer support? Are we still doing that?

3.2.0 is the version of the firrtl spec that added colored probes. I will add a feature guard to this PR.

> I'm unsure about starting by supporting parsing something until it's more robustly supported internally -- WDYT?

Personally I am OK with delivering changes "front to back", as long as there is a plan to finish up the work, which I think we have :). 

I keep thinking, maybe we should have an "enable experimental features" cli option in firtool, that we could use here for a bit of extra sanity while layers are in development (especially considering, the FIRRTL version for this feature is not the unstable/next version)

Reply:
Let's just own this through to the end, sounds good guys thanks!